### PR TITLE
Improve test coverage for file format of file data ingestion

### PR DESCRIPTION
**Pull request by olukas**
Checklist:
- [x] Labels and Milestone set
- [N/A] Added a line in `hazelcast-jet-distribution/src/root/release_notes.txt` (for any non-trivial fix/enhancement/feature)
- [N/A] New public APIs have `@Nonnull/@Nullable` annotations
- [N/A] New public APIs have `@since` tags in Javadoc
- [N/A] Updated `examples/README.md` (when adding a new code sample)

### CODE REVIEW

File: extensions/hadoop/src/test/java/com/hazelcast/jet/hadoop/file/ParquetFileFormatTest.java [hunk 1]
```diff
@@ -53,11 +62,73 @@ public void shouldReadParquetFile() throws Exception {
...
+                                                             .glob("file.parquet")
+                                                             .format(FileFormat.parquet());
+
+        if (useHadoop) {
+            source.useHadoopForLocalFiles(true);
+        }
+
+        Pipeline p = Pipeline.create();
+
+        // we need to used read object to fail the job
```
Comment (frant-hartm) [inline extensions/hadoop/src/test/java/com/hazelcast/jet/hadoop/file/ParquetFileFormatTest.java:99]: Better:
```suggestion
The type of the objects in the pipeline is not enforced at runtime. Adding a mapping step causes explicit cast, resulting in a failed job.
```

**Review by frant-hartm — APPROVED** [review]
LGTM. One minor thing with the comment.

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/3728/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/3728/jet-oss-pr-builder-3728.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:checkstyle (default) on project hazelcast-jet-hadoop: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There is 1 error reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/extensions/hadoop/src/test/java/com/hazelcast/jet/hadoop/file/ParquetFileFormatTest.java:100: Line has trailing spaces. [RegexpSingleline]
--------------------------
</details>
</pre>

**Comment by olukas** [comment]
verify

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/3761/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/3761/jet-oss-pr-builder-3761.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:checkstyle (default) on project hazelcast-jet-hadoop: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There is 1 error reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/extensions/hadoop/src/test/java/com/hazelcast/jet/hadoop/file/ParquetFileFormatTest.java:100: Line has trailing spaces. [RegexpSingleline]
--------------------------
</details>
</pre>

**Pull request merged**